### PR TITLE
Swap size and number arguments for calloc calls

### PR DESCRIPTION
**Pull request by AtariDreams**
Calloc's documentation says that the number of elements come first, not the size. The size comes after.

### CODE REVIEW

**Comment by igalic** [comment]
is this correcting all the calls to calloc()?
or just a small minority?
(if it's all of them, you might as well just fix the docs)

File: cddl/contrib/opensolaris/tools/ctf/cvt/merge.c [hunk 1]
```diff
@@ -748,7 +748,7 @@ conjure_function(tdesc_t *old, int newselfid, merge_cb_data_t *mcd)
 	tdesc_t *new = conjure_template(old, newselfid);
 	fndef_t *nfn = xmalloc(sizeof (fndef_t));
 	fndef_t *ofn = old->t_fndef;
-	int i;
+	uint_t i;
```
Comment (emaste) [inline cddl/contrib/opensolaris/tools/ctf/cvt/merge.c:751]: This is an unrelated change, and ought to go to https://github.com/openzfs/zfs

File: contrib/elftoolchain/libdwarf/libdwarf_init.c [hunk 2]
```diff
@@ -302,7 +302,7 @@ _dwarf_alloc(Dwarf_Debug *ret_dbg, int mode, Dwarf_Error *error)
 {
 	Dwarf_Debug dbg;
 
-	if ((dbg = calloc(sizeof(struct _Dwarf_Debug), 1)) == NULL) {
+	if ((dbg = calloc(1, sizeof(struct _Dwarf_Debug))) == NULL) {
```
Comment (emaste) [inline contrib/elftoolchain/libdwarf/libdwarf_init.c:305]: likewise this should go to https://sourceforge.net/projects/elftoolchain/

Comment (AtariDreams) [reply contrib/elftoolchain/libdwarf/libdwarf_init.c:305]: Fixed!

File: cddl/contrib/opensolaris/tools/ctf/cvt/merge.c (hunk 1 above)
Comment (AtariDreams) [reply cddl/contrib/opensolaris/tools/ctf/cvt/merge.c:751]: Fixed!

Comment (AtariDreams) [reply cddl/contrib/opensolaris/tools/ctf/cvt/merge.c]: Though I don't think you posted exactly the right upstream...

Comment (emaste) [reply cddl/contrib/opensolaris/tools/ctf/cvt/merge.c]: Ah, right you are.

**Comment by AtariDreams** [comment]
Fixed!

**Comment by ngie-eign** [comment]
NAK:
- Please submit all 3rd party fixes to their upstreams as requested by @emaste .
- Please change the email address to a non-users.noreply.github.com based address.
If you use `git config user.email`, etc, then run `git commit --reset-author --amend`, it will update the authorship appropriately.

**Comment by AtariDreams** [comment]
> NAK:
> 
> * Please submit all 3rd party fixes to their upstreams as requested by @emaste .
> * Please change the email address to a non-users.noreply.github.com based address.
>   If you use `git config user.email`, etc, then run `git commit --reset-author --amend`, it will update the authorship appropriately.
> * Please reformat the commit message to wrap all lines after the commit summary at <80 columns.

Done!

File: crypto/openssh/cipher.c [hunk 3]
```diff
@@ -255,7 +255,7 @@ cipher_init(struct sshcipher_ctx **ccp, const struct sshcipher *cipher,
 #endif
 
 	*ccp = NULL;
-	if ((cc = calloc(sizeof(*cc), 1)) == NULL)
+	if ((cc = calloc(1, sizeof(*cc))) == NULL)
```
Comment (bsdimp) [inline crypto/openssh/cipher.c:258]: this is contrib software and this change should be made upstream.

**Comment by bsdimp** [comment]
This has been in changes requested state for a while. If the changes are resolved it can be resubmitted.
But really, it's just a churn for churn sake's change. It needs to have every one of these utilities tested against the possibility of regression and submitted in a more granular fashion that we can make progress one.
